### ISSUE TITLE
Incorrect? atom types in SDF -> Mol2 conversion (O.3 vs O.2)

### ISSUE THREAD

**Issue by AlexW12344** [openbabel/openbabel#2610]
## Environment Information
Open Babel version: 3.1.0 
Operating system and version: Ubuntu 22.04.2 LTS

## Expected Behavior
Conversion of SDF to mol2 should give O.3 oxygen atom type for atom 43

## Actual Behavior
Conversion of SDF to mol2 gives O.2 oxygen atom type for atom 43, I think this is not correct based on the valance.

## Steps to Reproduce
SDF: 
```
26
                    3D
 Structure written by MMmdl.
 43 46  0  0  1  0            999 V2000
    3.8290  -49.9082    7.3567 O   0  0  0  0  0  0
    3.1153  -49.0416    6.8012 C   0  0  0  0  0  0
    3.5513  -47.8977    6.5562 O   0  5  0  0  0  0
    1.7057  -49.3580    6.4403 C   0  0  0  0  0  0
   -0.1076  -50.6487    6.3793 C   0  0  0  0  0  0
   -1.0211  -51.7084    6.5402 C   0  0  0  0  0  0
   -2.3348  -51.5531    6.0492 C   0  0  0  0  0  0
   -2.6993  -50.3586    5.3936 C   0  0  0  0  0  0
   -1.7635  -49.3099    5.2420 C   0  0  0  0  0  0
   -0.4423  -49.4338    5.7433 C   0  0  0  0  0  0
    0.7328  -48.6071    5.8015 C   0  0  0  0  0  0
    0.8592  -47.2049    5.2675 C   0  0  0  0  0  0
    0.9701  -46.1519    6.3924 C   0  0  0  0  0  0
   -0.0852  -45.0395    6.3276 C   0  0  0  0  0  0
   -1.4118  -45.4805    6.0441 O   0  0  0  0  0  0
   -2.2386  -45.8040    7.0949 C   0  0  0  0  0  0
   -4.0387  -46.5572    9.1587 C   0  0  0  0  0  0
   -2.7241  -47.0517    9.1502 C   0  0  0  0  0  0
   -1.8288  -46.6752    8.1339 C   0  0  0  0  0  0
   -0.7241  -52.6191    7.0419 H   0  0  0  0  0  0
   -3.0568  -52.3474    6.1618 H   0  0  0  0  0  0
   -3.7027  -50.2338    5.0158 H   0  0  0  0  0  0
   -2.0623  -48.3873    4.7602 H   0  0  0  0  0  0
    0.0186  -46.9829    4.6116 H   0  0  0  0  0  0
    1.7413  -47.1366    4.6230 H   0  0  0  0  0  0
    1.9528  -45.6851    6.3456 H   0  0  0  0  0  0
    0.9550  -46.6329    7.3658 H   0  0  0  0  0  0
    0.1913  -44.3765    5.5164 H   0  0  0  0  0  0
   -0.0677  -44.4348    7.2308 H   0  0  0  0  0  0
   -0.8492  -47.1142    8.1369 H   0  0  0  0  0  0
   -4.4644  -45.6869    8.1312 C   0  0  0  0  0  0
   -3.5686  -45.3157    7.0918 C   0  0  0  0  0  0
   -4.0196  -44.4491    6.0697 C   0  0  0  0  0  0
   -5.3473  -43.9766    6.0646 C   0  0  0  0  0  0
   -6.2357  -44.3575    7.0895 C   0  0  0  0  0  0
   -5.7910  -45.2055    8.1244 C   0  0  0  0  0  0
   -3.3603  -44.1405    5.2743 H   0  0  0  0  0  0
   -5.6744  -43.3185    5.2717 H   0  0  0  0  0  0
   -7.2548  -43.9991    7.0843 H   0  0  0  0  0  0
   -6.4744  -45.4962    8.9093 H   0  0  0  0  0  0
   -2.4004  -47.7370    9.9189 H   0  0  0  0  0  0
   -4.7163  -46.8570    9.9410 H   0  0  0  0  0  0
    1.2056  -50.5983    6.7955 O   0  0  0  0  0  0
  1  2  2  0  0  0
  2  3  1  0  0  0
  2  4  1  0  0  0
  4 11  2  0  0  0
  4 43  1  0  0  0
  5  6  1  0  0  0
  5 10  2  0  0  0
  5 43  1  0  0  0
  6  7  2  0  0  0
  6 20  1  0  0  0
  7  8  1  0  0  0
  7 21  1  0  0  0
  8  9  2  0  0  0
  8 22  1  0  0  0
  9 10  1  0  0  0
  9 23  1  0  0  0
 10 11  1  0  0  0
 11 12  1  0  0  0
 12 24  1  0  0  0
 12 25  1  0  0  0
 12 13  1  0  0  0
 13 14  1  0  0  0
 13 26  1  0  0  0
 13 27  1  0  0  0
 14 15  1  0  0  0
 14 28  1  0  0  0
 14 29  1  0  0  0
 15 16  1  0  0  0
 16 19  2  0  0  0
 16 32  1  0  0  0
 17 18  2  0  0  0
 17 31  1  0  0  0
 17 42  1  0  0  0
 18 19  1  0  0  0
 18 41  1  0  0  0
 19 30  1  0  0  0
 31 32  2  0  0  0
 31 36  1  0  0  0
 32 33  1  0  0  0
 33 34  2  0  0  0
 33 37  1  0  0  0
 34 35  1  0  0  0
 34 38  1  0  0  0
 35 36  2  0  0  0
 35 39  1  0  0  0
 36 40  1  0  0  0
M  CHG  1   3  -1
M  END
```
cmd line to run: 
```
obabel -i sdf input1.sdf -o mol2 -O out.mol2 --partialcharge none -m
```
Output mol2:
```
@<TRIPOS>MOLECULE
26
 43 46 0 0 0
SMALL
GASTEIGER
****
Structure written by MMmdl.
@<TRIPOS>ATOM
      1 O           3.8290  -49.9082    7.3567 O.co2   1  UNL1        0.0000
      2 C           3.1153  -49.0416    6.8012 C.2     1  UNL1        0.0000
      3 O           3.5513  -47.8977    6.5562 O.co2   1  UNL1        0.0000
      4 C           1.7057  -49.3580    6.4403 C.ar    1  UNL1        0.0000
      5 C          -0.1076  -50.6487    6.3793 C.ar    1  UNL1        0.0000
      6 C          -1.0211  -51.7084    6.5402 C.ar    1  UNL1        0.0000
      7 C          -2.3348  -51.5531    6.0492 C.ar    1  UNL1        0.0000
      8 C          -2.6993  -50.3586    5.3936 C.ar    1  UNL1        0.0000
      9 C          -1.7635  -49.3099    5.2420 C.ar    1  UNL1        0.0000
     10 C          -0.4423  -49.4338    5.7433 C.ar    1  UNL1        0.0000
     11 C           0.7328  -48.6071    5.8015 C.ar    1  UNL1        0.0000
     12 C           0.8592  -47.2049    5.2675 C.3     1  UNL1        0.0000
     13 C           0.9701  -46.1519    6.3924 C.3     1  UNL1        0.0000
     14 C          -0.0852  -45.0395    6.3276 C.3     1  UNL1        0.0000
     15 O          -1.4118  -45.4805    6.0441 O.3     1  UNL1        0.0000
     16 C          -2.2386  -45.8040    7.0949 C.ar    1  UNL1        0.0000
     17 C          -4.0387  -46.5572    9.1587 C.ar    1  UNL1        0.0000
     18 C          -2.7241  -47.0517    9.1502 C.ar    1  UNL1        0.0000
     19 C          -1.8288  -46.6752    8.1339 C.ar    1  UNL1        0.0000
     20 H          -0.7241  -52.6191    7.0419 H       1  UNL1        0.0000
     21 H          -3.0568  -52.3474    6.1618 H       1  UNL1        0.0000
     22 H          -3.7027  -50.2338    5.0158 H       1  UNL1        0.0000
     23 H          -2.0623  -48.3873    4.7602 H       1  UNL1        0.0000
     24 H           0.0186  -46.9829    4.6116 H       1  UNL1        0.0000
     25 H           1.7413  -47.1366    4.6230 H       1  UNL1        0.0000
     26 H           1.9528  -45.6851    6.3456 H       1  UNL1        0.0000
     27 H           0.9550  -46.6329    7.3658 H       1  UNL1        0.0000
     28 H           0.1913  -44.3765    5.5164 H       1  UNL1        0.0000
     29 H          -0.0677  -44.4348    7.2308 H       1  UNL1        0.0000
     30 H          -0.8492  -47.1142    8.1369 H       1  UNL1        0.0000
     31 C          -4.4644  -45.6869    8.1312 C.ar    1  UNL1        0.0000
     32 C          -3.5686  -45.3157    7.0918 C.ar    1  UNL1        0.0000
     33 C          -4.0196  -44.4491    6.0697 C.ar    1  UNL1        0.0000
     34 C          -5.3473  -43.9766    6.0646 C.ar    1  UNL1        0.0000
     35 C          -6.2357  -44.3575    7.0895 C.ar    1  UNL1        0.0000
     36 C          -5.7910  -45.2055    8.1244 C.ar    1  UNL1        0.0000
     37 H          -3.3603  -44.1405    5.2743 H       1  UNL1        0.0000
     38 H          -5.6744  -43.3185    5.2717 H       1  UNL1        0.0000
     39 H          -7.2548  -43.9991    7.0843 H       1  UNL1        0.0000
     40 H          -6.4744  -45.4962    8.9093 H       1  UNL1        0.0000
     41 H          -2.4004  -47.7370    9.9189 H       1  UNL1        0.0000
     42 H          -4.7163  -46.8570    9.9410 H       1  UNL1        0.0000
     43 O           1.2056  -50.5983    6.7955 O.2     1  UNL1        0.0000
@<TRIPOS>UNITY_ATOM_ATTR
3 1
charge -1
@<TRIPOS>BOND
     1     1     2   ar
     2     2     3   ar
     3     2     4    1
     4     4    11   ar
     5     4    43   ar
     6     5     6   ar
     7     5    10   ar
     8     5    43   ar
     9     6     7   ar
    10     6    20    1
    11     7     8   ar
    12     7    21    1
    13     8     9   ar
    14     8    22    1
    15     9    10   ar
    16     9    23    1
    17    10    11   ar
    18    11    12    1
    19    12    24    1
    20    12    25    1
    21    12    13    1
    22    13    14    1
    23    13    26    1
    24    13    27    1
    25    14    15    1
    26    14    28    1
    27    14    29    1
    28    15    16    1
    29    16    19   ar
    30    16    32   ar
    31    17    18   ar
    32    17    31   ar
    33    17    42    1
    34    18    19   ar
    35    18    41    1
    36    19    30    1
    37    31    32   ar
    38    31    36   ar
    39    32    33   ar
    40    33    34   ar
    41    33    37    1
    42    34    35   ar
    43    34    38    1
    44    35    36   ar
    45    35    39    1
    46    36    40    1
```

**Comment by robertwerner67** [comment]
Atom 43 is the oxygen atom within the benzofuran moiety

![structure](https://github.com/openbabel/openbabel/assets/60490931/bee4a45d-f047-46f6-84d8-afe92f1a27a0)

for which one typically assigns sp2 hybridization.  The SYBYL atom type notation encodes this by `O.2`; the reference I use here is the table on page 53 of the mid-2005 published SYBYL 7.1/Tripos Mol2 documentation still available as a copy by the time machine on archive.org ([link](https://web.archive.org/web/20160615131623/http://www.tripos.com/data/support/mol2.pdf), record by 2016-06-15, see the file attached below).

----
Two sidenotes:

+ You are correct that sdf can be a container of multiple structures. Here, your use of `-m` to split the output into multiple output files however is not necessary because the input file is about a single model containing only one molecule.
+ You want to process an input file of syntax of a sdf file which actually carries the file extension `.sdf`. The intended output shall follow the mol2 syntax in a file with the file extension of `.mol2`; again coherent.  In such a case, the file extensions are typically *indicative  enough* for openbabel to recognize the details of the task ahead; you can issue a shorter instruction to yield the same result:

```shell
obabel input1.sdf -O out.mol2 --partialcharge none
```


[mol2.pdf](https://github.com/openbabel/openbabel/files/12025037/mol2.pdf)

**Comment by AlexW12344** [comment]
I see thats my bad with the chemistry. Thanks for the thorough answer and references.
